src/firefly/js/visualize/saga/CatalogWatcher.js

```diff
@@ -42,13 +42,15 @@ export function* watchCatalogs() {
 
 
     while (true) {
-        const action= yield take([TABLE_NEW_LOADED, TABLE_SORT, TABLE_SELECT,TABLE_HIGHLIGHT, TABLE_UPDATE,
+        const action= yield take([TABLE_LOADED, TABLE_SELECT,TABLE_HIGHLIGHT, TABLE_UPDATE,
                                   TABLE_REMOVE, ImagePlotCntlr.PLOT_IMAGE]);
         const {tbl_id}= action.payload;
         switch (action.type) {
-            case TABLE_SORT:
-            case TABLE_NEW_LOADED:
-                handleCatalogUpdate(tbl_id);
+            case TABLE_LOADED:
+                const {invokedBy} = action.payload;
+                if (invokedBy !== TABLE_SORT) {
```

Comment:
I mean that we do need to refetch catalog data on TABLE_SORT. We did it
before.

2016-08-02 15:34 GMT-07:00 Emmanuel Joliet <notifications@github.com>:

> In src/firefly/js/visualize/saga/CatalogWatcher.js
> <https://github.com/Caltech-IPAC/firefly/pull/129#discussion_r73250913>:
>
> >                                    TABLE_REMOVE, ImagePlotCntlr.PLOT_IMAGE]);
> >          const {tbl_id}= action.payload;
> >          switch (action.type) {
> > -            case TABLE_SORT:
> > -            case TABLE_NEW_LOADED:
> > -                handleCatalogUpdate(tbl_id);
> > +            case TABLE_LOADED:
> > +                const {invokedBy} = action.payload;
> > +                if (invokedBy !== TABLE_SORT) {
>
> before, when sorting, the table highlight was the first row. So it might
> not be a bug.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Caltech-IPAC/firefly/pull/129/files/cd8e95c74816df03bb70f93d1edd494d9d9bb6bc#r73250913>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALU3QJKBxw0C9hF-wcAlB54KvHtrR8Egks5qb8XqgaJpZM4JaF7x>
> .
>